frame/support/src/storage/types/key.rs

```diff
@@ -270,735 +270,22 @@ pub trait HasReversibleKeyPrefix<P>: ReversibleKeyGenerator + HasKeyPrefix<P> {
-	(M, N, O, P, Q, R)
-);
-impl_key_prefix_for!(
-	(A, B, C, D, E, F, G, H, I, J, K, L, M, N, O, P, Q, R),
-	(A, B, C, D, E, F, G, H, I, J, K),
-	(L, M, N, O, P, Q, R)
-);
-impl_key_prefix_for!(
-	(A, B, C, D, E, F, G, H, I, J, K, L, M, N, O, P, Q, R),
-	(A, B, C, D, E, F, G, H, I, J),
-	(K, L, M, N, O, P, Q, R)
-);
-impl_key_prefix_for!(
-	(A, B, C, D, E, F, G, H, I, J, K, L, M, N, O, P, Q, R),
-	(A, B, C, D, E, F, G, H, I),
-	(J, K, L, M, N, O, P, Q, R)
-);
-impl_key_prefix_for!(
-	(A, B, C, D, E, F, G, H, I, J, K, L, M, N, O, P, Q, R),
-	(A, B, C, D, E, F, G, H),
-	(I, J, K, L, M, N, O, P, Q, R)
-);
-impl_key_prefix_for!(
-	(A, B, C, D, E, F, G, H, I, J, K, L, M, N, O, P, Q, R),
-	(A, B, C, D, E, F, G),
-	(H, I, J, K, L, M, N, O, P, Q, R)
-);
-impl_key_prefix_for!(
-	(A, B, C, D, E, F, G, H, I, J, K, L, M, N, O, P, Q, R),
-	(A, B, C, D, E, F),
-	(G, H, I, J, K, L, M, N, O, P, Q, R)
-);
-impl_key_prefix_for!(
-	(A, B, C, D, E, F, G, H, I, J, K, L, M, N, O, P, Q, R),
-	(A, B, C, D, E),
-	(F, G, H, I, J, K, L, M, N, O, P, Q, R)
-);
-impl_key_prefix_for!(
-	(A, B, C, D, E, F, G, H, I, J, K, L, M, N, O, P, Q, R),
-	(A, B, C, D),
-	(E, F, G, H, I, J, K, L, M, N, O, P, Q, R)
-);
-impl_key_prefix_for!(
-	(A, B, C, D, E, F, G, H, I, J, K, L, M, N, O, P, Q, R),
-	(A, B, C),
-	(D, E, F, G, H, I, J, K, L, M, N, O, P, Q, R)
-);
-impl_key_prefix_for!(
-	(A, B, C, D, E, F, G, H, I, J, K, L, M, N, O, P, Q, R),
-	(A, B),
-	(C, D, E, F, G, H, I, J, K, L, M, N, O, P, Q, R)
-);
-impl_key_prefix_for!(
-	(A, B, C, D, E, F, G, H, I, J, K, L, M, N, O, P, Q, R),
-	A,
-	(B, C, D, E, F, G, H, I, J, K, L, M, N, O, P, Q, R)
+frame_support_procedural::impl_key_prefix_for!(A, B);
+frame_support_procedural::impl_key_prefix_for!(A, B, C);
+frame_support_procedural::impl_key_prefix_for!(A, B, C, D);
+frame_support_procedural::impl_key_prefix_for!(A, B, C, D, E);
+frame_support_procedural::impl_key_prefix_for!(A, B, C, D, E, F);
+frame_support_procedural::impl_key_prefix_for!(A, B, C, D, E, F, G);
+frame_support_procedural::impl_key_prefix_for!(A, B, C, D, E, F, G, H);
+frame_support_procedural::impl_key_prefix_for!(A, B, C, D, E, F, G, H, I);
+frame_support_procedural::impl_key_prefix_for!(A, B, C, D, E, F, G, H, I, J);
+frame_support_procedural::impl_key_prefix_for!(A, B, C, D, E, F, G, H, I, J, K);
+frame_support_procedural::impl_key_prefix_for!(A, B, C, D, E, F, G, H, I, J, K, L);
+frame_support_procedural::impl_key_prefix_for!(A, B, C, D, E, F, G, H, I, J, K, L, M);
+frame_support_procedural::impl_key_prefix_for!(A, B, C, D, E, F, G, H, I, J, K, L, M, N);
+frame_support_procedural::impl_key_prefix_for!(A, B, C, D, E, F, G, H, I, J, K, L, M, N, O);
+frame_support_procedural::impl_key_prefix_for!(A, B, C, D, E, F, G, H, I, J, K, L, M, N, O, P);
+frame_support_procedural::impl_key_prefix_for!(A, B, C, D, E, F, G, H, I, J, K, L, M, N, O, P, Q);
+frame_support_procedural::impl_key_prefix_for!(
+	A, B, C, D, E, F, G, H, I, J, K, L, M, N, O, P, Q, R,
 );
```

Comment:
This should be one macro call. This is way to verbose. Why do I need to pass here the identifiers? You can generate them.

This is also an internal macro that just serves the purpose of generating these impls. No need to make it that much configurable. Maximum could be that you pass the maximum number of identifiers to generate/